pkg/controller/instance/instance_controller.go

```diff
@@ -404,21 +473,36 @@ func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Resu
 			instance.Spec.OperatorVersion.Name,
 			instance.Name,
 			err)
-		r.recorder.Event(instance, "Warning", "InvalidOperatorVersion", fmt.Sprintf("Error getting operatorversion \"%v\": %v", ov.Name, err))
-		return reconcile.Result{}, err
+		if r != nil {
```

Comment:
why would r be nil? 🤔

Reply:
details written on method comment above.   the function allows for the recorder to be nil.  This was to support getting OV in the predicate and reconcile.   It did not make sense to record an event to an instance that did not exist because the create event loop was stopped.

The expectation is that r will be nil when getting an OV where an instance doesn't exist yet... otherwise it is a recorder.